corehq/pillows/case_search.py

```diff
@@ -0,0 +1,151 @@
+import inspect
+import re
+
+from casexml.apps.case.models import CommCareCase
+from corehq.apps.case_search.exceptions import CaseSearchNotEnabledException
+from corehq.apps.case_search.models import case_search_enabled_for_domain
+from corehq.apps.change_feed import topics
+from corehq.apps.change_feed.consumer.feed import KafkaChangeFeed
+from corehq.elastic import get_es_new
+from corehq.pillows.case import CASE_ES_TYPE, CasePillow
+from corehq.pillows.const import CASE_SEARCH_ALIAS
+from corehq.pillows.mappings.case_search_mapping import CASE_SEARCH_INDEX, \
+    CASE_SEARCH_MAPPING
+from corehq.util.couch_helpers import paginate_view
+from pillowtop.checkpoints.manager import PillowCheckpoint, \
+    PillowCheckpointEventHandler
+from pillowtop.es_utils import ElasticsearchIndexMeta
+from pillowtop.feed.couch import change_from_couch_row
+from pillowtop.feed.interface import Change
+from pillowtop.pillow.interface import ConstructedPillow
+from pillowtop.processors.elastic import ElasticProcessor
+from pillowtop.reindexer.change_providers.couch import CouchViewChangeProvider
+from pillowtop.reindexer.reindexer import PillowReindexer
+
+
+class CaseSearchPillow(CasePillow):
+    # TODO: Remove this once elasticsearch gets bootstrapped with new pillows
+    # (Used in order to set up index)
+    """
+    Nested case properties indexer.
+    """
+    es_alias = CASE_SEARCH_ALIAS
+
+    es_index = CASE_SEARCH_INDEX
+    default_mapping = CASE_SEARCH_MAPPING
+
+    def change_trigger(self, changes_dict):
+        return
+
+
+def transform_case_for_elasticsearch(doc_dict):
+    doc = {
+        desired_property: doc_dict.get(desired_property)
+        for desired_property in CASE_SEARCH_MAPPING['properties'].keys()
+        if desired_property != 'case_properties'
+    }
+    doc['_id'] = doc_dict.get('_id')
+    doc['case_properties'] = _get_case_properties(doc_dict)
+    return doc
+
+
+def _get_case_properties(doc_dict):
```

Comment:
interesting, i'm not sure there is an easy way to go from the dict to a properly wrapped model. it would be nice if there was a standardized way to do this from the dicts themselves. maybe we can start that here and have a todo that it needs to be updated to deal with SQL better?